src/runtime/virtcontainers/pkg/agent/protocols/client/client.go

```diff
@@ -312,7 +312,8 @@ func parseGrpcHybridVSockAddr(sock string) (string, uint32, error) {
 // https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
 func commonDialer(timeout time.Duration, dialFunc func() (net.Conn, error), timeoutErrMsg error) (net.Conn, error) {
 	t := time.NewTimer(timeout)
-	cancel := make(chan bool)
+	// Do not block, just notify the sub-goroutin to exit
```

Comment:
lgtm. Maybe you can change this comment to "Create a buffered channel so that notifying subroutine is not blocked"